File: MyLeetCodeC++/stack.cpp
#include <iostream>
using namespace std;

class Stack
{
private:
    int arr, top, size;
public:
    Stack()
    {
        this->arr = [1];
        this->top = -1;
        this->size = 0;
    };
    void push(int x)
    {
        if (top == (size - 1))
        {
            cout << "overflow";
        }
        size++;
        top++;
        arr[top] = x;
        return;
    }


   void print()
    {
        int size = top;
        if (top == -1)
        {
            cout << "empty";
        }
        while (size != -1)
        {
            cout << arr[size];
            size--;
        }
    return;
    }
};

int
main()
{
    Stack stk;
    stk.print();
    return 0;
}

File: Concept/stack.js
class Stack {
  constructor(length) {
    this.item = [];
    this.top = -1;
    this.stkLen = length;
  }

  push(elem) {
    if (top == this.stkLen - 1) {
      return "overflow";
    }
    this.top++;
    return this.item.push(elem);
  }
  pop() {
    if (this.top == -1) {
      return "underflow";
    }
    this.top--;
    return this.item.pop();
  }
  Top() {
    return this.top;
  }
  empty() {
    return this.top == -1;
  }
}

const stack = new Stack();

stack.push(5);
console.log(stack.top);
